plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt

```diff
@@ -37,59 +37,66 @@ import org.jetbrains.kotlin.utils.addToStdlib.safeAs
 
 object DefaultDescriptorToDocumentableTranslator : SourceToDocumentableTranslator {
 
-    override fun invoke(platformData: PlatformData, context: DokkaContext): DModule {
+    override fun invoke(sourceSets: SourceSetData, context: DokkaContext): DModule {
 
-        val (environment, facade) = context.platforms.getValue(platformData)
+        val (environment, facade) = context.platforms.getValue(sourceSets)
         val packageFragments = environment.getSourceFiles().asSequence()
             .map { it.packageFqName }
             .distinct()
             .mapNotNull { facade.resolveSession.getPackageFragment(it) }
             .toList()
 
-        return DokkaDescriptorVisitor(platformData, context.platforms.getValue(platformData).facade, context.logger).run {
+        return DokkaDescriptorVisitor(sourceSets, context.platforms.getValue(sourceSets).facade, context.logger).run {
             packageFragments.mapNotNull { it.safeAs<PackageFragmentDescriptor>() }.map {
                 visitPackageFragmentDescriptor(
                     it,
-                    DRIWithPlatformInfo(DRI.topLevel, PlatformDependent.empty())
+                    DRIWithPlatformInfo(DRI.topLevel, emptyMap())
                 )
             }
-        }.let { DModule(platformData.name, it, PlatformDependent.empty(), listOf(platformData)) }
+        }.let { DModule(sourceSets.moduleName, it, emptyMap(), null, listOf(sourceSets)) }
     }
 }
 
 data class DRIWithPlatformInfo(
     val dri: DRI,
-    val actual: PlatformDependent<DocumentableSource>
+    val actual: SourceSetDependent<DocumentableSource>
 )
 
-fun DRI.withEmptyInfo() = DRIWithPlatformInfo(this, PlatformDependent.empty())
+fun DRI.withEmptyInfo() = DRIWithPlatformInfo(this, emptyMap())
 
 private class DokkaDescriptorVisitor(
-    private val platformData: PlatformData,
+    private val sourceSet: SourceSetData,
     private val resolutionFacade: DokkaResolutionFacade,
     private val logger: DokkaLogger
 ) : DeclarationDescriptorVisitorEmptyBodies<Documentable, DRIWithPlatformInfo>() {
     override fun visitDeclarationDescriptor(descriptor: DeclarationDescriptor, parent: DRIWithPlatformInfo): Nothing {
         throw IllegalStateException("${javaClass.simpleName} should never enter ${descriptor.javaClass.simpleName}")
     }
 
+    private fun Collection<DeclarationDescriptor>.filterDescriptorsInSourceSet() = filter {
+            val path = it.toSourceElement.containingFile.toString()
+            path != null && sourceSet.sourceRoots.any { root -> path.startsWith(root.path) }
```

Comment:
```suggestion
            path.isNotBlank() && sourceSet.sourceRoots.any { root -> path.startsWith(root.path) }
```

`path != null` is always true